tests/test_autogen_diffs.py

```diff
@@ -705,6 +711,111 @@ def test_integer(self):
+            Column("number", Numeric()),
+            Column("dec", DECIMAL()),
+            Column("timestamp", TIMESTAMP()),
+            Column("datetime", DateTime()),
+            Column("flag", Boolean()),
+            Column("bigint", BigInteger()),
+            Column("smallint", SmallInteger()),
+            Column("date", DATE()),
+            Column("string", String(32)),
+            Column("largebin", LargeBinary()),
+            Column("unicode", Unicode(32)),
+            Column("enum", Enum("one", "two", "three", name="the_enum")),
+        )
+        return m
+
+    @classmethod
+    def _get_model_schema(cls):
+        m = MetaData()
+        Table(
+            "sometable",
+            m,
+            Column("id", INTEGER(), primary_key=True),
+            Column("char", CHAR()),
+            Column("varchar", VARCHAR(32)),
+            Column("txt", Text()),
+            Column("float", FLOAT()),
+            Column("number", Numeric()),
+            Column("dec", DECIMAL()),
+            Column("timestamp", TIMESTAMP()),
+            Column("datetime", DateTime()),
+            Column("flag", Boolean()),
+            Column("bigint", BigInteger()),
+            Column("smallint", SmallInteger()),
+            Column("date", DATE()),
+            Column("string", String(32)),
+            Column("largebin", LargeBinary()),
+            Column("unicode", Unicode(32)),
+            Column("enum", Enum("one", "two", "three", name="the_enum")),
+        )
+        return m
+
+    def test_introspected_columns_match_metadata_columns(self):
+        uo = ops.UpgradeOps(ops=[])
+        autogenerate._produce_net_changes(self.autogen_context, uo)
+        diffs = uo.as_diffs()
+        eq_(diffs, [])
+
+
+class PickUpStringTypeChangeTest(AutogenTest, TestBase):
+    @classmethod
+    def _get_db_schema(cls):
+        m = MetaData()
+        Table(
+            "sometable",
+            m,
+            Column("id", INTEGER(), primary_key=True),
+            Column("stringcol", Text),
+        )
+        return m
+
+    @classmethod
+    def _get_model_schema(cls):
+        m = MetaData()
+        Table(
+            "sometable",
+            m,
+            Column("id", INTEGER(), primary_key=True),
+            Column("stringcol", String(255)),
+        )
+        return m
+
+    def test_picks_up_changed_type(self):
+        uo = ops.UpgradeOps(ops=[])
+        autogenerate._produce_net_changes(self.autogen_context, uo)
+        diffs = uo.as_diffs()
+
+        eq_(diffs[0][0][0], "modify_type")
+        eq_(diffs[0][0][3], "stringcol")
+
+
+class ComparePGTZDatetimeSpecificityTest(TestBase):
+    __only_on__ = "postgresql"
+
+    def _fixture(self):
+        from alembic.ddl.postgresql import PostgresqlImpl
+        from sqlalchemy.dialects import postgresql
+
+        return PostgresqlImpl(
+            postgresql.dialect(), None, False, True, None, {}
+        )
+
     def test_datetime(self):
```

Comment:
**mike bayer** (zzzeek) wrote:

Interesting, that you need the PG dialect now.  that is fine.   However, I think the test that is here should also be in the SQLite version, where we assert that the compare_type is False in all cases.  this gets into the bigger issue of comparing types that have dialect-specific flags and those that use Variants.  if look at AutogenerateVariantCompareTest in this same file we can probably add some more fun tests with things like `String().with_variant(mysql.VARCHAR(collation='xyz'), "mysql")`.

Reply:
Now that you point it out, probably didn't actually need that here. Wrote this test when the code was much less general. Let me have a look.